src/jit/liveness.cpp

```diff
@@ -2251,6 +2251,9 @@ void Compiler::fgComputeLifeLIR(VARSET_TP& life, BasicBlock* block, VARSET_VALAR
             case GT_RETURNTRAP:
             case GT_PUTARG_STK:
             case GT_IL_OFFSET:
+#ifdef FEATURE_HW_INTRINSICS
+            case GT_HWIntrinsic:
```

Comment:
Question. Shall we check intrinsic id or category to only avoid removing Store* (other HW intrinsics still can be removed)?

Reply:
Do we want to be able to optimize away user code (for HWIntrinsics)?

I think, for the most part, we should always emit it as is and can assume the user has hand optimized it.

@CarolEidt, thoughts?

Reply:
I had a similar thought in looking at this, and while I agree that one might generally assume the user has hand-optimized, we may find that a method using HW intrinsics is inlined into another method that passes constant parameters or otherwise renders some of the code unreachable or unused. So, I think it would be useful to at least consider having a method on `GenTreeHWIntrinsic` that indicates whether it is side-effecting. We'd probably want it to have a default value, and unfortunately the conservative default value would be `true`, but the ones for which it would really need to be true are probably a small percentage of the intrinsics.

Reply:
Is it fine to log a work item to track this or do you want me to plumb the GTF_SIDE_EFFECT flag through the HWIntrinsic store instructions?

Reply:
(I'm fine with either, just wanting to know where I should start the work).

Reply:
I think it would be best to file an issue/work item to track this. The incremental benefit of enabling optimizing away unused or unreached intrinsics is outweighed by getting them implemented and specifically (in this case) gaining understanding of how it works to directly implement the stores as intrinsics.

Reply:
https://github.com/dotnet/coreclr/issues/16088